### PR TITLE
Feature scalebar plugin

### PR DESCRIPTION
**Pull request by Burak-Atak**
-Add scale bar checkbox
-Add scale bar to cached settings
-Update options for image processing
-Base scale bar plugin created, will be implemented new functionality

### CODE REVIEW

File: uscope/imagep/streams.py [hunk 1]
```diff
@@ -364,6 +365,10 @@ def run(self):
         current_image = current_images[0]
 
         ipp = config.get_usc().ipp.pipeline_last()
+        current_plugins = [p["plugin"] for p in ipp]
```
Comment (JohnDMcMaster) [inline uscope/imagep/streams.py:368]: This is not philosophically how the current system works. Instead the intention is that config only contains a simple data structure (think JSON) and that configuration drives which plugins are required.

File: uscope/app/argus/widgets.py [hunk 2]
```diff
@@ -1005,14 +1008,26 @@ def getProgressLayout():
...
 
         layout.addLayout(getNameLayout())
 
         self.snapshot_pb = QPushButton("Snap")
         self.snapshot_pb.setIcon(QIcon(config.GUI.icon_files["camera"]))
         self.snapshot_pb.clicked.connect(self.take_snapshot)
-        layout.addWidget(self.snapshot_pb)
+        snap_layout.addWidget(self.snapshot_pb)
 
+        add_scalebar_label = QLabel("Add scalebar to image")
```
Comment (JohnDMcMaster) [inline uscope/app/argus/widgets.py:1020]: Long term we might want to rethink this to keep UI simple (menu option?) but this work for MVP

**Comment by JohnDMcMaster** [comment]
Thanks for your WIP PR! My main feddback is to pass in configuration to drive the plugin pipeline creation instead of passing in explicit plugins. Please let me know if that helps guide this and/or you have any additional questions

File: uscope/imagep/streams.py (hunk 1 above)
Comment (Burak-Atak) [reply uscope/imagep/streams.py:368]: Yes, I couldn't be sure about, here's logic, I should analyse the project and rethink for here

File: uscope/app/argus/widgets.py (hunk 2 above)
Comment (Burak-Atak) [reply uscope/app/argus/widgets.py:1020]: I can change the widget type, I put this checkbox according to the documentation

**Comment by Burak-Atak** [comment]
Thank you for your feedback, I will refactor pr and update here very soon, If I have any question I will ask thank you.

File: uscope/imagep/streams.py (hunk 1 above)
Comment (Burak-Atak) [reply uscope/imagep/streams.py:368]: Hello @JohnDMcMaster, I added the plugin functionality. I did not understand how exactly should I implement this part. I wrote this code according to https://docs.google.com/document/d/18Lcze718SK4ZbZlFLauQnipJrSRh27BItK7SEN_bmbw/edit this documentation, If you could explain your request to me, I can change the code according to you.

File: fonts/arial.ttf
Comment (JohnDMcMaster) [inline fonts/arial.ttf:1]: What is the license on this file? Where did it come from?

Comment (yusufm) [reply fonts/arial.ttf:1]: @Burak-Atak please use an open source font. There should be enough options, maybe a fixed width font? Courier?

Comment (yusufm) [reply fonts/arial.ttf:1]: There is a large free collection of fonts at https://developers.google.com/fonts/faq#is_there_a_cost_for_any_of_these_fonts

Comment (Burak-Atak) [reply fonts/arial.ttf:1]: I added the roboto font family from google fonts instead of arial font

**Comment by JohnDMcMaster** [comment]
Thanks for the patch! I think this is good enough to merge in and we can make some tweaks post merge as needed. Notes:
- Font path is relative to the module, but does not use a relative path. This causes the code to crash as written. This will need to be fixed for it to be usable
- As previously noted we need to move the scalebar option somewhere else, location TBD
- Scale is hard to read as is with many nm. Might reformat with um / mm / etc like we do on the measure page
- Labsmore.com and measurement are pressed together. I'm going to space them apart

**Pull request merged**